README.md

```diff
@@ -3,30 +3,233 @@
+  },
+})
+```
+
+Take a look at [TLS create secure context](https://nodejs.org/dist/latest-v8.x/docs/api/tls.html#tls_tls_createsecurecontext_options) for more information.
+
+#### <a name="setup-client-sasl"></a> SASL
+
+Kafka has support for using SASL to authenticate clients. The `sasl` option can be used to configure the authentication mechanism. Currently, KafkaJS only supports the `PLAIN` mechanism.
+
+```javascript
+new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  sasl: {
+    mechanism: 'plain',
+    username: 'my-username',
+    password: 'my-password'
+  },
+})
+```
+
+It is __highly recommended__ that you use SSL for encryption when using `PLAIN`.
+
+#### <a name="setup-client-connection-timeout"></a> Connection Timeout
+
+Time in milliseconds to wait for a successful connection. The default value is: `1000`.
+
+```javascript
+new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  connectionTimeout: 3000
+})
+```
+
+#### <a name="setup-client-default-retry"></a> Default Retry
+
+The `retry` option can be used to set the default configuration. The retry mechanism uses a randomization function that grows exponentially.
+
+Available options:
+
+* __maxRetryTime__ - Maximum wait time for a retry in milliseconds. Default: `30000`
+* __initialRetryTime__ - Initial value used to calculate the retry in milliseconds (This is still randomized following the randomization factor). Default: `300`
+* __factor__ - Randomization factor. Default: `0.2`
+* __multiplier__ - Exponential factor. Default: `2`
+* __retries__ - Max number of retries per call. Default: `5`
+
+```javascript
+new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  retry: {
+    initialRetryTime: 100,
+    retries: 8
+  }
+})
+```
+
+#### <a name="setup-client-logger"></a> Logger
+
+KafkaJS has a built-in `STDOUT` logger which outputs JSON. It also accepts a custom log creator which allows you to integrate your favorite logger library.
+There are 5 log levels available: `NOTHING`, `ERROR`, `WARN`, `INFO`, and `DEBUG`. `INFO` is configured by default.
+
+__How to configure the log level?__
+
+```javascript
+const { Kafka, logLevel } = require('kafkajs')
+
+// Create the client with the broker list
+const kafka = new Kafka({
+  clientId: 'my-app',
+  brokers: ['kafka1:9092', 'kafka2:9092'],
+  logLevel: logLevel.ERROR
+})
+```
+
+The environment variable `KAFKAJS_LOG_LEVEL` can also be used and it has precedence over the configuration in code, example:
+
+```sh
+KAFKAJS_LOG_LEVEL=info node code.js
+```
+
+__How to create a log creator?___
+
+A log creator is a function which receives a log level and returns a log function. The log function receives two arguments: namespace and log.
+
+`namespace` identifies the component which is performing the log, for example, connection or consumer.
+
+`log` is an object with the following keys: `level`, `timestamp`, 'logger', and `message`
```

Comment:
you did use a apostrophe around "logger" instead of an accent. I'm not sure why and which effect that has...

Reply:
done